src/containers/DirectSum.h

```diff
@@ -144,8 +144,10 @@ class DirectSum : public ParticleContainer<Particle, ParticleCell> {
     return false;
   }
 
-  ParticleIterator<Particle, ParticleCell> begin(IteratorBehavior behavior = IteratorBehavior::haloAndOwned) override {
-    return ParticleIterator<Particle, ParticleCell>(&this->_data, &_cellBorderFlagManager, behavior);
+  ParticleIteratorWrapper<Particle> begin(IteratorBehavior behavior = IteratorBehavior::haloAndOwned) override {
+    return ParticleIteratorWrapper<Particle>(
+        (ParticleIteratorInterface<Particle> *)new internal::ParticleIterator<Particle, ParticleCell>(
```

Comment:
is this cast necessary? In the respective LinkedCell version, there is none and g++ compiles this without a cast.

Reply:
not necessary, since I have added that in the constructor